packages/fulfillment/integration-tests/__tests__/fulfillment-module-service.spec.ts

```diff
@@ -249,6 +249,200 @@ moduleIntegrationTestRunner({
+              shipping_profile_id: shippingProfile.id,
+              service_provider_id: providerId,
+              type: {
+                code: "test-type",
+                description: "test-description",
+                label: "test-label",
+              },
+              data: {
+                amount: 1000,
+              },
+            }
+
+            const [shippingOption1] = await service.createShippingOptions([
+              {
+                name: "test-option",
+                price_type: "flat",
+                ...commonShippingOptionData,
+                rules: [
+                  {
+                    attribute: "test-attribute",
+                    operator: "in",
+                    value: ["test-value"],
+                  },
+                ],
+              },
+              {
+                name: "test-option-2",
+                price_type: "flat",
+                ...commonShippingOptionData,
+                rules: [
+                  {
+                    attribute: "test-attribute2",
+                    operator: "eq",
+                    value: "test",
+                  },
+                ],
+              },
+            ])
+
+            const listedOptions = await service.listShippingOptions({
+              name: shippingOption1.name,
+            })
+
+            expect(listedOptions).toEqual(
+              expect.arrayContaining([
+                expect.objectContaining({ id: shippingOption1.id }),
+              ])
+            )
+          })
+
+          it("should list shipping options with a context", async function () {
+            const fulfillmentSet = await service.create({
+              name: "test",
+              type: "test-type",
+              service_zones: [
+                {
+                  name: "test",
+                },
+              ],
+            })
+
+            const shippingProfile = await service.createShippingProfiles({
+              name: "test",
+              type: "default",
+            })
+
+            // TODO: change that for a real provider instead of fake data manual inserted data
+            const [{ id: providerId }] =
+              await MikroOrmWrapper.forkManager().execute(
+                "insert into service_provider (id) values ('sp_jdafwfleiwuonl') returning id"
+              )
+
+            const commonShippingOptionData = {
+              service_zone_id: fulfillmentSet.service_zones[0].id,
+              shipping_profile_id: shippingProfile.id,
+              service_provider_id: providerId,
+              type: {
+                code: "test-type",
+                description: "test-description",
+                label: "test-label",
+              },
+              data: {
+                amount: 1000,
+              },
+            }
+
+            const [shippingOption1, ,shippingOption3] = await service.createShippingOptions([{
+              name: "test-option",
```

Comment:
We can create some fixtures so we don't have to repeat the same creation options again and again. Then if you want to change them, just destructure and change only the fields you care about.